hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestBackgroundContainerDataScanner.java

```diff
@@ -175,20 +176,19 @@ public void testUnhealthyContainerNotRescanned() throws Exception {
         .setState(UNHEALTHY);
     // Update the mock to reflect this.
     when(unhealthy.getContainerState()).thenReturn(UNHEALTHY);
-    assertFalse(unhealthy.shouldScanData());
-
-    // Clear metrics to check the next run.
-    metrics.resetNumContainersScanned();
-    metrics.resetNumUnhealthyContainers();
+    assertTrue(unhealthy.shouldScanData());
 
+    when(controller.markContainerUnhealthy(eq(unhealthy.getContainerData().getContainerID()),
+        any())).thenReturn(false);
     scanner.runIteration();
-    // The only invocation of unhealthy on this container should have been from
-    // the previous scan.
-    verifyContainerMarkedUnhealthy(unhealthy, atMostOnce());
-    // This iteration should skip the already unhealthy container.
+    // The invocation of unhealthy on this container will also happen in the
+    // next iteration.
+    verifyContainerMarkedUnhealthy(unhealthy, atMost(2));
```

Comment:
I think this should be exactly once.

Reply:
Same in other scanners' tests.

Reply:
I will still be called. It just exits early out of the function and returns `false`, so we don't update the `numUnhealthyContainers`

Reply:
Oh got it. I was thinking this was for `KeyValueHandler`, not `ContainerController`.